helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -360,18 +364,36 @@ public void addJob(final Job job) throws JobExistsException {
   }
 
   /**
-   * Given a jobId, returns the N most recent events in it's history in the cluster.
+   * Given a jobId, returns the N most recent events in its history in the cluster.
    */
   @Override
   public List<TaskStatusEvent> getJobHistory(final JobId jobId) throws JobDoesNotExistException {
+    return getJobHistory(jobId, null);
+  }
+
+  /**
+   * Given a jobId and host, returns the N most recent events in its history on that host in the
+   * cluster.
+   */
+  @Override
+  public List<TaskStatusEvent> getJobHistory(final JobId jobId, final String host)
+      throws JobDoesNotExistException {
     final Job descriptor = getJob(jobId);
     if (descriptor == null) {
       throw new JobDoesNotExistException(jobId);
     }
     final ZooKeeperClient client = provider.get("getJobHistory");
-    final List<String> hosts;
+    List<String> hosts;
```

Comment:
This can become `final` again if you decide to use the approach in the next comment.